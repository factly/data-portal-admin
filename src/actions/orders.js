import axios from '../utils/axios';
import {
  baseUrl,
  ADD_ORDERS_LIST_REQUEST,
  ADD_ORDER_DETAILS_REQUEST,
  LOADING_ORDERS,
  LOADING_ORDER_DETAILS,
  LOADING_ORDER_ITEMS,
  SET_ORDERS_LIST_TOTAL,
  SET_ORDERS_LIST_CURRENT_PAGE,
  LOAD_ORDERS_SUCCESS,
  LOAD_ORDERS_FAILURE,
  GET_ORDER_DETAILS_SUCCESS,
  GET_ORDER_DETAILS_FAILURE,
  GET_ORDER_ITEMS_SUCCESS,
  SET_ORDER_ITEMS_LIST_TOTAL,
  SET_ORDER_DETAILS_CURRENT_PAGE,
  GET_ORDER_ITEMS_FAILURE,
} from '../constants/orders';
import { loadCartsSuccess } from './carts';
import { loadPaymentsSuccess } from './payments';
import { addCurrencies } from './currencies';
import { loadProductsSuccess } from './products';
import { getIds, getValues, deleteKeys, buildObjectOfItems } from '../utils/objects';

export const loadOrders = (page = 1, limit) => {
  return async (dispatch, getState) => {
    let url = baseUrl;
    if (page && limit) {
      url = `${url}?page=${page}&limit=${limit}`;
    }

    const {
      orders: {
        list: { req },
      },
    } = getState();

    let found = false;
    let ids;
    for (let item of req) {
      if (item.page === page && item.limit === limit) {
        ids = [...item.ids];
        found = true;
      }
    }

    if (found) {
      dispatch(setListCurrentPage(ids));
      return;
    }

    dispatch(loadingOrders());

    const response = await axios({
      url: url,
      method: 'get',
    }).catch((error) => {
      dispatch(loadOrdersFailure(error.message));
    });

    if (response) {
      const { nodes, total } = response.data;

      const carts = getValues(nodes, 'cart');
      dispatch(loadCartsSuccess(carts));

      const payments = getValues(nodes, 'payment');
      const currencies = getValues(payments, 'currency');
      dispatch(addCurrencies(currencies));
      dispatch(loadPaymentsSuccess(payments));

      const currentPageIds = getIds(nodes);
      const req = { page: page, limit: limit, ids: currentPageIds };
      dispatch(addListRequest(req));
      dispatch(loadOrdersSuccess(nodes));
      dispatch(setListCurrentPage(currentPageIds));
      dispatch(setOrdersListTotal(total));
    }
  };
};

export const getOrderDetails = (id) => {
  return async (dispatch, getState) => {
    let url = `${baseUrl}/${id}`;

    dispatch(loadingOrderDetails());

    const response = await axios({
      url: url,
      method: 'get',
    }).catch((error) => {
      dispatch(getOrderDetailsFailure(error.message));
    });

    if (response) {
      dispatch(getOrderDetailsSuccess(response.data));
    }
  };
};

export const getOrderItems = (id, page = 1, limit) => {
  return async (dispatch, getState) => {
    let url = `${baseUrl}/${id}/items`;
    if (page && limit) {
      url = `${url}?page=${page}&limit=${limit}`;
    }

    dispatch(loadingOrderItems());

    const {
      orders: {
        details: { req },
      },
    } = getState();

    let reqObj;
    for (let item of req) {
      if (item.orderId === id && item.page === page && item.limit === limit) {
        reqObj = item;
        break;
      }
    }

    if (reqObj) {
      dispatch(setDetailsCurrentPage(reqObj.ids));
      dispatch(setOrderItemsListTotal(reqObj.total));
      return;
    }

    const response = await axios({
      url: url,
      method: 'get',
    }).catch((error) => {
      dispatch(getOrderItemsFailure(error.message));
    });

    if (response) {
      const { nodes, total } = response.data;

      const products = getValues(nodes, 'product');
      const currencies = getValues(products, 'currency');
      dispatch(addCurrencies(currencies));
      dispatch(loadProductsSuccess(products));

      const currentPageIds = getIds(nodes);
      const req = { orderId: id, page, limit, ids: currentPageIds, total };
      dispatch(addDetailsRequest(req));
      dispatch(getOrderItemsSuccess(nodes));
      dispatch(setDetailsCurrentPage(currentPageIds));
      dispatch(setOrderItemsListTotal(total));
    }
  };
};

const setListCurrentPage = (ids) => {
  return {
    type: SET_ORDERS_LIST_CURRENT_PAGE,
    payload: ids,
  };
};

const setDetailsCurrentPage = (ids) => {
  return {
    type: SET_ORDER_DETAILS_CURRENT_PAGE,
    payload: ids,
  };
};

const addListRequest = (req) => {
  return {
    type: ADD_ORDERS_LIST_REQUEST,
    payload: req,
  };
};

const addDetailsRequest = (req) => {
  return {
    type: ADD_ORDER_DETAILS_REQUEST,
    payload: req,
  };
};

const loadingOrders = () => {
  return {
    type: LOADING_ORDERS,
  };
};

const loadingOrderDetails = () => {
  return {
    type: LOADING_ORDER_DETAILS,
  };
};

const loadingOrderItems = () => {
  return {
    type: LOADING_ORDER_ITEMS,
  };
};

const setOrdersListTotal = (total) => {
  return {
    type: SET_ORDERS_LIST_TOTAL,
    payload: total,
  };
};

const setOrderItemsListTotal = (total) => {
  return {
    type: SET_ORDER_ITEMS_LIST_TOTAL,
    payload: total,
  };
};

const loadOrdersSuccess = (orders) => {
  return {
    type: LOAD_ORDERS_SUCCESS,
    payload: {
      items: buildObjectOfItems(deleteKeys(orders, ['cart', 'payment'])),
    },
  };
};

const loadOrdersFailure = (message) => {
  return {
    type: LOAD_ORDERS_FAILURE,
    payload: message,
  };
};

const getOrderDetailsSuccess = (order) => {
  return {
    type: GET_ORDER_DETAILS_SUCCESS,
    payload: order,
  };
};

const getOrderDetailsFailure = (message) => {
  return {
    type: GET_ORDER_DETAILS_FAILURE,
    payload: message,
  };
};

const getOrderItemsSuccess = (orderItems) => {
  return {
    type: GET_ORDER_ITEMS_SUCCESS,
    payload: {
      items: buildObjectOfItems(deleteKeys(orderItems, ['product'])),
    },
  };
};

const getOrderItemsFailure = (message) => {
  return {
    type: GET_ORDER_ITEMS_FAILURE,
    payload: message,
  };
};
