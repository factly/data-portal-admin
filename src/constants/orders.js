export const LOADING_ORDERS = 'LOADING_ORDERS';
export const LOADING_ORDER_DETAILS = 'LOADING_ORDER_DETAILS';
export const LOADING_ORDER_ITEMS = 'LOADING_ORDER_ITEMS';
export const LOAD_ORDERS_SUCCESS = 'LOAD_ORDERS_SUCCESS';
export const LOAD_ORDERS_FAILURE = 'LOAD_ORDERS_FAILURE';
export const GET_ORDER_DETAILS_SUCCESS = 'GET_ORDER_DETAILS_SUCCESS';
export const GET_ORDER_DETAILS_FAILURE = 'GET_ORDER_DETAILS_FAILURE';
export const GET_ORDER_ITEMS_SUCCESS = 'GET_ORDER_ITEMS_SUCCESS';
export const GET_ORDER_ITEMS_FAILURE = 'GET_ORDER_ITEMS_FAILURE';

export const baseUrl = `${process.env.REACT_APP_API_URL}/orders`;
