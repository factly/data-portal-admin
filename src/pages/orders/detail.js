import React, { useState } from 'react';
import { useParams } from "react-router-dom";
import { Table, Form, Descriptions, Card } from 'antd';


const Orders = () => {
  const [form] = Form.useForm();
  const [data, setData] = useState([]);
  const [total, setTotal] = useState(0);
  const { id } = useParams();

  React.useEffect(() => {
    fetch(`${process.env.REACT_APP_API_URL}/orders/${id}/items`)
      .then((data) => data.json())
      .then((data) => {
        setData(data.nodes);
        setTotal(data.total);
      });
  }, [id]);

  const get = (page, limit) => {
    fetch(`${process.env.REACT_APP_API_URL}/orders/${id}/items`)
      .then((data) => data.json())
      .then((data) => {
        setData(data.nodes);
        setTotal(data.total);
      });
  };

  const columns = [
    {
      title: 'Product Item',
      render: (record) => record.Product.title,
      width: '40%',
    },
    {
      title: 'Price',
      render: (record) => <span>{record.Product.price} {record.Product.Currency.iso_code}</span>,
      width: '20%',
    },
    {
      title: 'Extra Info',
      dataIndex: 'extra_info',
      width: '30%',
    },
  ];


  const Order = data.length && data[0].Order;

  return (
    <div>
      {Order && (
        <Card>
        <Descriptions title="Order details">
          <Descriptions.Item label="Order ID">{Order.ID}</Descriptions.Item>
          <Descriptions.Item label="Order Status">{Order.status}</Descriptions.Item>
          <Descriptions.Item label="Payment">{Order.Payment.amount} {Order.Payment.Currency.iso_code}</Descriptions.Item>
          <Descriptions.Item label="Payment Gateway">{Order.Payment.gateway}</Descriptions.Item>
          <Descriptions.Item label="Payment Status">{Order.Payment.status}</Descriptions.Item>
        </Descriptions>
        </Card>
      )}

      <Form form={form} component={false}>
        <Table
          bordered
          rowKey="ID"
          dataSource={data}
          columns={columns}
          pagination={{
            onChange: get,
            total: total,
          }}
        />
      </Form>
    </div>
  );
};

export default Orders;
