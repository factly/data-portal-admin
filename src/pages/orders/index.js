import React, { useState } from 'react';
import { Table, Form, Button } from 'antd';
import moment from 'moment';


const Orders = (props) => {
  const [form] = Form.useForm();
  const [data, setData] = useState([]);
  const [total, setTotal] = useState(0);

  React.useEffect(() => {
    fetch(process.env.REACT_APP_API_URL + '/orders')
      .then((data) => data.json())
      .then((data) => {
        setData(data.nodes);
        setTotal(data.total);
      });
  }, []);

  const get = (page, limit) => {
    fetch(process.env.REACT_APP_API_URL + '/orders?page=' + page + '&limit=' + limit)
      .then((data) => data.json())
      .then((data) => {
        setData(data.nodes);
        setTotal(data.total);
      });
  };

  const columns = [
    {
      title: 'Cart',
      dataIndex: 'cart_id',
      width: '20%',
    },
    {
      title: 'Amount',
      render: (record) => <span>{`${record.Payment.amount} ${record.Payment.Currency.iso_code}`}</span>,
      width: '20%',
    },
    {
      title: 'User',
      dataIndex: 'user_id',
      width: '20%',
    },
    {
      title: 'Status',
      dataIndex: 'status',
      width: '20%',
    },
    {
      title: 'Created At',
      dataIndex: 'CreatedAt',
      width: '20%',
      render: (_, record) => {
        return <span title={record.CreatedAt}>{moment(record.CreatedAt).fromNow()}</span>;
      },
    },
    {
      title: 'Operation',
      dataIndex: 'operation',
      render: (_, record) => {
        return (
        <span>
            <Button
              type="primary"
              onClick={() => props.history.push(`${process.env.PUBLIC_URL}/orders/${record.ID}`)}
              style={{
                marginRight: 8,
              }}
              >
              Details
            </Button>
        </span>
        )
      }
    }
  ];

  return (
    <div>
      <Form form={form} component={false}>
        <Table
          bordered
          rowKey="ID"
          dataSource={data}
          columns={columns}
          pagination={{
            onChange: get,
            total: total,
          }}
        />
      </Form>
    </div>
  );
};

export default Orders;
