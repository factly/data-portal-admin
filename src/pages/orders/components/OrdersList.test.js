import React from 'react';
import { Router } from 'react-router-dom';
import renderer, { act } from 'react-test-renderer';
import { useDispatch, Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { mount } from 'enzyme';
import { Button, Table } from 'antd';

import '../../../matchMedia.mock';
import OrderList from './OrdersList';
import { loadOrders } from '../../../actions/orders';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useDispatch: jest.fn(),
}));
jest.mock('../../../actions/orders', () => ({
  loadOrders: jest.fn(),
  deleteOrder: jest.fn(),
}));

describe('Orders List component', () => {
  let store;
  let mockedDispatch;
  const order = {
    id: 1,
    cart_id: 1,
    user_id: 1,
    status: 'status',
    payment_id: 1,
    created_at: '2020-12-12',
  };

  describe('snapshot testing', () => {
    beforeEach(() => {
      store = mockStore({
        orders: {
          ids: [1],
          total: 1,
          req: [
            {
              page: 1,
              limit: 5,
              ids: [1],
            },
          ],
          items: {
            1: {
              id: 1,
              cart_id: 1,
              user_id: 1,
              status: 'status',
              payment_id: 1,
              created_at: '2020-12-12',
            },
          },
        },
        payments: {
          items: {
            1: {
              id: 1,
              currency_id: 1,
              amount: 100,
            },
          },
        },
        currencies: {
          items: {
            1: {
              id: 1,
              iso_code: 'INR',
            },
          },
        },
      });
      store.dispatch = jest.fn(() => ({}));
      mockedDispatch = jest.fn();
      useDispatch.mockReturnValue(mockedDispatch);
    });
    it('should render the component', () => {
      const tree = renderer
        .create(
          <Provider store={store}>
            <OrderList />
          </Provider>,
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('should match component when loading', () => {
      store = mockStore({
        orders: {
          ids: [],
          total: 0,
          loading: true,
          req: [],
          items: {},
        },
        payments: {},
        currencies: {},
      });
      const tree = renderer
        .create(
          <Provider store={store}>
            <OrderList />
          </Provider>,
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('should match component with orders', () => {
      loadOrders.mockReset();

      let component;
      act(() => {
        component = renderer.create(
          <Provider store={store}>
            <OrderList />
          </Provider>,
        );
      });
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
      expect(mockedDispatch).toHaveBeenCalledTimes(1);
      expect(loadOrders).toHaveBeenCalledWith(1, 5);
    });
  });
  describe('component testing', () => {
    beforeEach(() => {
      jest.clearAllMocks();
      store = mockStore({
        orders: {
          ids: [1],
          total: 1,
          req: [
            {
              page: 1,
              limit: 5,
              ids: [1],
            },
          ],
          items: {
            1: {
              id: 1,
              cart_id: 1,
              user_id: 1,
              status: 'status',
              payment_id: 1,
              created_at: '2020-12-12',
            },
          },
        },
        payments: {
          items: {
            1: {
              id: 1,
              currency_id: 1,
              amount: 100,
            },
          },
        },
        currencies: {
          items: {
            1: {
              id: 1,
              iso_code: 'INR',
            },
          },
        },
      });
      mockedDispatch = jest.fn(() => new Promise((resolve) => resolve(true)));
      useDispatch.mockReturnValue(mockedDispatch);
    });
    it('should change the page', () => {
      const wrapper = mount(
        <Provider store={store}>
          <OrderList />
        </Provider>,
      );
      const table = wrapper.find(Table);
      table.props().pagination.onChange(2);
      wrapper.update();
      const updatedTable = wrapper.find(Table);
      expect(updatedTable.props().pagination.current).toEqual(2);
    });
    it('should go to order details page', () => {
      const historyMock = { push: jest.fn(), location: {}, listen: jest.fn() };
      const wrapper = mount(
        <Provider store={store}>
          <Router history={historyMock}>
            <OrderList />
          </Router>
        </Provider>,
      );
      const button = wrapper.find(Button).at(0);
      expect(button.text()).toEqual('Details');
      button.simulate('click');

      expect(historyMock.push).toHaveBeenCalledWith('/orders/1');
    });
    it('should not have details buttons', () => {
      store = mockStore({
        orders: {
          ids: [],
          total: 0,
          loading: true,
          req: [],
          items: {},
        },
        payments: {},
        currencies: {},
      });
      const wrapper = mount(
        <Provider store={store}>
          <OrderList />
        </Provider>,
      );

      const button = wrapper.find(Button);
      expect(button.length).toEqual(0);
    });
  });
});
