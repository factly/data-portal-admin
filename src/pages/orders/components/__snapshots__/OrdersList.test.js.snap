// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Orders List component snapshot testing should match component when loading 1`] = `
<div>
  <div
    className="ant-table-wrapper"
  >
    <div
      className="ant-spin-nested-loading"
    >
      <div
        className="ant-spin-container"
      >
        <div
          className="ant-table ant-table-bordered ant-table-empty"
        >
          <div
            className="ant-table-container"
          >
            <div
              className="ant-table-content"
              onScroll={[Function]}
              style={Object {}}
            >
              <table
                style={
                  Object {
                    "tableLayout": "auto",
                  }
                }
              >
                <colgroup>
                  <col
                    style={
                      Object {
                        "minWidth": "20%",
                        "width": "20%",
                      }
                    }
                  />
                  <col
                    style={
                      Object {
                        "minWidth": "20%",
                        "width": "20%",
                      }
                    }
                  />
                  <col
                    style={
                      Object {
                        "minWidth": "20%",
                        "width": "20%",
                      }
                    }
                  />
                  <col
                    style={
                      Object {
                        "minWidth": "20%",
                        "width": "20%",
                      }
                    }
                  />
                  <col
                    style={
                      Object {
                        "minWidth": "20%",
                        "width": "20%",
                      }
                    }
                  />
                </colgroup>
                <thead
                  className="ant-table-thead"
                >
                  <tr>
                    <th
                      className="ant-table-cell"
                      colSpan={null}
                      rowSpan={null}
                      style={Object {}}
                    >
                      Cart
                    </th>
                    <th
                      className="ant-table-cell"
                      colSpan={null}
                      rowSpan={null}
                      style={Object {}}
                    >
                      Amount
                    </th>
                    <th
                      className="ant-table-cell"
                      colSpan={null}
                      rowSpan={null}
                      style={Object {}}
                    >
                      User
                    </th>
                    <th
                      className="ant-table-cell"
                      colSpan={null}
                      rowSpan={null}
                      style={Object {}}
                    >
                      Status
                    </th>
                    <th
                      className="ant-table-cell"
                      colSpan={null}
                      rowSpan={null}
                      style={Object {}}
                    >
                      Created At
                    </th>
                    <th
                      className="ant-table-cell"
                      colSpan={null}
                      rowSpan={null}
                      style={Object {}}
                    >
                      Operation
                    </th>
                  </tr>
                </thead>
                <tbody
                  className="ant-table-tbody"
                >
                  <tr
                    className="ant-table-placeholder"
                    style={
                      Object {
                        "display": null,
                      }
                    }
                  >
                    <td
                      className="ant-table-cell"
                      colSpan={6}
                      rowSpan={null}
                      style={Object {}}
                    >
                      <div
                        className="ant-empty ant-empty-normal"
                      >
                        <div
                          className="ant-empty-image"
                        >
                          <svg
                            className="ant-empty-img-simple"
                            height="41"
                            viewBox="0 0 64 41"
                            width="64"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <g
                              fill="none"
                              fillRule="evenodd"
                              transform="translate(0 1)"
                            >
                              <ellipse
                                className="ant-empty-img-simple-ellipse"
                                cx="32"
                                cy="33"
                                rx="32"
                                ry="7"
                              />
                              <g
                                className="ant-empty-img-simple-g"
                                fillRule="nonzero"
                              >
                                <path
                                  d="M55 12.76L44.854 1.258C44.367.474 43.656 0 42.907 0H21.093c-.749 0-1.46.474-1.947 1.257L9 12.761V22h46v-9.24z"
                                />
                                <path
                                  className="ant-empty-img-simple-path"
                                  d="M41.613 15.931c0-1.605.994-2.93 2.227-2.931H55v18.137C55 33.26 53.68 35 52.05 35h-40.1C10.32 35 9 33.259 9 31.137V13h11.16c1.233 0 2.227 1.323 2.227 2.928v.022c0 1.605 1.005 2.901 2.237 2.901h14.752c1.232 0 2.237-1.308 2.237-2.913v-.007z"
                                />
                              </g>
                            </g>
                          </svg>
                        </div>
                        <p
                          className="ant-empty-description"
                        >
                          No Data
                        </p>
                      </div>
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Orders List component snapshot testing should match component with orders 1`] = `
<div>
  <div
    className="ant-table-wrapper"
  >
    <div
      className="ant-spin-nested-loading"
    >
      <div
        className="ant-spin-container"
      >
        <div
          className="ant-table ant-table-bordered"
        >
          <div
            className="ant-table-container"
          >
            <div
              className="ant-table-content"
              onScroll={[Function]}
              style={Object {}}
            >
              <table
                style={
                  Object {
                    "tableLayout": "auto",
                  }
                }
              >
                <colgroup>
                  <col
                    style={
                      Object {
                        "minWidth": "20%",
                        "width": "20%",
                      }
                    }
                  />
                  <col
                    style={
                      Object {
                        "minWidth": "20%",
                        "width": "20%",
                      }
                    }
                  />
                  <col
                    style={
                      Object {
                        "minWidth": "20%",
                        "width": "20%",
                      }
                    }
                  />
                  <col
                    style={
                      Object {
                        "minWidth": "20%",
                        "width": "20%",
                      }
                    }
                  />
                  <col
                    style={
                      Object {
                        "minWidth": "20%",
                        "width": "20%",
                      }
                    }
                  />
                </colgroup>
                <thead
                  className="ant-table-thead"
                >
                  <tr>
                    <th
                      className="ant-table-cell"
                      colSpan={null}
                      rowSpan={null}
                      style={Object {}}
                    >
                      Cart
                    </th>
                    <th
                      className="ant-table-cell"
                      colSpan={null}
                      rowSpan={null}
                      style={Object {}}
                    >
                      Amount
                    </th>
                    <th
                      className="ant-table-cell"
                      colSpan={null}
                      rowSpan={null}
                      style={Object {}}
                    >
                      User
                    </th>
                    <th
                      className="ant-table-cell"
                      colSpan={null}
                      rowSpan={null}
                      style={Object {}}
                    >
                      Status
                    </th>
                    <th
                      className="ant-table-cell"
                      colSpan={null}
                      rowSpan={null}
                      style={Object {}}
                    >
                      Created At
                    </th>
                    <th
                      className="ant-table-cell"
                      colSpan={null}
                      rowSpan={null}
                      style={Object {}}
                    >
                      Operation
                    </th>
                  </tr>
                </thead>
                <tbody
                  className="ant-table-tbody"
                >
                  <tr
                    className="ant-table-row ant-table-row-level-0"
                    data-row-key={1}
                    onClick={[Function]}
                    style={Object {}}
                  >
                    <td
                      className="ant-table-cell"
                      colSpan={null}
                      rowSpan={null}
                      style={Object {}}
                    >
                      1
                    </td>
                    <td
                      className="ant-table-cell"
                      colSpan={null}
                      rowSpan={null}
                      style={Object {}}
                    >
                      <span>
                        100 INR
                      </span>
                    </td>
                    <td
                      className="ant-table-cell"
                      colSpan={null}
                      rowSpan={null}
                      style={Object {}}
                    >
                      1
                    </td>
                    <td
                      className="ant-table-cell"
                      colSpan={null}
                      rowSpan={null}
                      style={Object {}}
                    >
                      status
                    </td>
                    <td
                      className="ant-table-cell"
                      colSpan={null}
                      rowSpan={null}
                      style={Object {}}
                    >
                      <span
                        title="2020-12-12"
                      >
                        4 months ago
                      </span>
                    </td>
                    <td
                      className="ant-table-cell"
                      colSpan={null}
                      rowSpan={null}
                      style={Object {}}
                    >
                      <span>
                        <button
                          className="ant-btn ant-btn-primary"
                          onClick={[Function]}
                          style={
                            Object {
                              "marginRight": 8,
                            }
                          }
                          type="button"
                        >
                          <span>
                            Details
                          </span>
                        </button>
                      </span>
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
        </div>
        <ul
          className="ant-pagination ant-table-pagination ant-table-pagination-right"
          style={Object {}}
          unselectable="unselectable"
        >
          <li
            aria-disabled={true}
            className="ant-pagination-prev ant-pagination-disabled"
            onClick={[Function]}
            onKeyPress={[Function]}
            tabIndex={null}
            title="Previous Page"
          >
            <button
              className="ant-pagination-item-link"
              disabled={true}
              tabIndex={-1}
              type="button"
            >
              <span
                aria-label="left"
                className="anticon anticon-left"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  className=""
                  data-icon="left"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
                  />
                </svg>
              </span>
            </button>
          </li>
          <li
            className="ant-pagination-item ant-pagination-item-1 ant-pagination-item-active"
            onClick={[Function]}
            onKeyPress={[Function]}
            tabIndex="0"
            title={1}
          >
            <a
              rel="nofollow"
            >
              1
            </a>
          </li>
          <li
            aria-disabled={true}
            className="ant-pagination-next ant-pagination-disabled"
            onClick={[Function]}
            onKeyPress={[Function]}
            tabIndex={null}
            title="Next Page"
          >
            <button
              className="ant-pagination-item-link"
              disabled={true}
              tabIndex={-1}
              type="button"
            >
              <span
                aria-label="right"
                className="anticon anticon-right"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  className=""
                  data-icon="right"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
                  />
                </svg>
              </span>
            </button>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;

exports[`Orders List component snapshot testing should render the component 1`] = `
<div>
  <div
    className="ant-table-wrapper"
  >
    <div
      className="ant-spin-nested-loading"
    >
      <div
        className="ant-spin-container"
      >
        <div
          className="ant-table ant-table-bordered"
        >
          <div
            className="ant-table-container"
          >
            <div
              className="ant-table-content"
              onScroll={[Function]}
              style={Object {}}
            >
              <table
                style={
                  Object {
                    "tableLayout": "auto",
                  }
                }
              >
                <colgroup>
                  <col
                    style={
                      Object {
                        "minWidth": "20%",
                        "width": "20%",
                      }
                    }
                  />
                  <col
                    style={
                      Object {
                        "minWidth": "20%",
                        "width": "20%",
                      }
                    }
                  />
                  <col
                    style={
                      Object {
                        "minWidth": "20%",
                        "width": "20%",
                      }
                    }
                  />
                  <col
                    style={
                      Object {
                        "minWidth": "20%",
                        "width": "20%",
                      }
                    }
                  />
                  <col
                    style={
                      Object {
                        "minWidth": "20%",
                        "width": "20%",
                      }
                    }
                  />
                </colgroup>
                <thead
                  className="ant-table-thead"
                >
                  <tr>
                    <th
                      className="ant-table-cell"
                      colSpan={null}
                      rowSpan={null}
                      style={Object {}}
                    >
                      Cart
                    </th>
                    <th
                      className="ant-table-cell"
                      colSpan={null}
                      rowSpan={null}
                      style={Object {}}
                    >
                      Amount
                    </th>
                    <th
                      className="ant-table-cell"
                      colSpan={null}
                      rowSpan={null}
                      style={Object {}}
                    >
                      User
                    </th>
                    <th
                      className="ant-table-cell"
                      colSpan={null}
                      rowSpan={null}
                      style={Object {}}
                    >
                      Status
                    </th>
                    <th
                      className="ant-table-cell"
                      colSpan={null}
                      rowSpan={null}
                      style={Object {}}
                    >
                      Created At
                    </th>
                    <th
                      className="ant-table-cell"
                      colSpan={null}
                      rowSpan={null}
                      style={Object {}}
                    >
                      Operation
                    </th>
                  </tr>
                </thead>
                <tbody
                  className="ant-table-tbody"
                >
                  <tr
                    className="ant-table-row ant-table-row-level-0"
                    data-row-key={1}
                    onClick={[Function]}
                    style={Object {}}
                  >
                    <td
                      className="ant-table-cell"
                      colSpan={null}
                      rowSpan={null}
                      style={Object {}}
                    >
                      1
                    </td>
                    <td
                      className="ant-table-cell"
                      colSpan={null}
                      rowSpan={null}
                      style={Object {}}
                    >
                      <span>
                        100 INR
                      </span>
                    </td>
                    <td
                      className="ant-table-cell"
                      colSpan={null}
                      rowSpan={null}
                      style={Object {}}
                    >
                      1
                    </td>
                    <td
                      className="ant-table-cell"
                      colSpan={null}
                      rowSpan={null}
                      style={Object {}}
                    >
                      status
                    </td>
                    <td
                      className="ant-table-cell"
                      colSpan={null}
                      rowSpan={null}
                      style={Object {}}
                    >
                      <span
                        title="2020-12-12"
                      >
                        4 months ago
                      </span>
                    </td>
                    <td
                      className="ant-table-cell"
                      colSpan={null}
                      rowSpan={null}
                      style={Object {}}
                    >
                      <span>
                        <button
                          className="ant-btn ant-btn-primary"
                          onClick={[Function]}
                          style={
                            Object {
                              "marginRight": 8,
                            }
                          }
                          type="button"
                        >
                          <span>
                            Details
                          </span>
                        </button>
                      </span>
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
        </div>
        <ul
          className="ant-pagination ant-table-pagination ant-table-pagination-right"
          style={Object {}}
          unselectable="unselectable"
        >
          <li
            aria-disabled={true}
            className="ant-pagination-prev ant-pagination-disabled"
            onClick={[Function]}
            onKeyPress={[Function]}
            tabIndex={null}
            title="Previous Page"
          >
            <button
              className="ant-pagination-item-link"
              disabled={true}
              tabIndex={-1}
              type="button"
            >
              <span
                aria-label="left"
                className="anticon anticon-left"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  className=""
                  data-icon="left"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
                  />
                </svg>
              </span>
            </button>
          </li>
          <li
            className="ant-pagination-item ant-pagination-item-1 ant-pagination-item-active"
            onClick={[Function]}
            onKeyPress={[Function]}
            tabIndex="0"
            title={1}
          >
            <a
              rel="nofollow"
            >
              1
            </a>
          </li>
          <li
            aria-disabled={true}
            className="ant-pagination-next ant-pagination-disabled"
            onClick={[Function]}
            onKeyPress={[Function]}
            tabIndex={null}
            title="Next Page"
          >
            <button
              className="ant-pagination-item-link"
              disabled={true}
              tabIndex={-1}
              type="button"
            >
              <span
                aria-label="right"
                className="anticon anticon-right"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  className=""
                  data-icon="right"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
                  />
                </svg>
              </span>
            </button>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
